### PR TITLE
feat(chord): allow downloading DATS JSON as attachment

### PR DESCRIPTION
**Pull request by davidlougheed**

### CODE REVIEW

File: chord_metadata_service/restapi/utils.py [hunk 1]
```diff
@@ -88,3 +107,23 @@ def build_experiments_by_subject(biosamples_experiments_details: list[dict]) ->
...
+
+def response_as_attachment(request: DrfRequest) -> bool:
+    """
+    Helper function for processing the as_attachment query parameter, for consistent behaviour when returning JSON
+    responses as file attachments.
+    """
+    return request.query_params.get("attachment", "").strip().lower() in ("1", "true", "yes")
+
+
+FILENAME_REPLACE_PATTERN = re.compile(r"[\\/:*?\"<>|]")
```
Comment (v-rocheleau) [inline chord_metadata_service/restapi/utils.py]: Put with other constant

File: chord_metadata_service/chord/tests/test_api.py [hunk 2]
```diff
@@ -184,6 +184,24 @@ def test_dats(self):
         response = self.client.get("/api/datasets/does-not-exist/dats")
         self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
 
+    def test_dats_as_attachment(self):
```
Comment (v-rocheleau) [inline chord_metadata_service/chord/tests/test_api.py:187]: Would add a query without the `?attachment` URL param as well

File: chord_metadata_service/patients/api_views.py [hunk 3]
```diff
@@ -132,13 +134,10 @@ def phenopackets(self, request, *_args, **_kwargs):
             .order_by("id")
         )
 
-        filename_safe_id = re.sub(r"[\\/:*?\"<>|]", "_", individual.id)
-        return Response(
+        return response_optionally_as_attachment(
```
Comment (v-rocheleau) [inline chord_metadata_service/patients/api_views.py:137]: Is this already covered in tests?

**Review by v-rocheleau — CHANGES_REQUESTED** [review]
minor requests, otherwise lgtm

File: chord_metadata_service/patients/api_views.py (hunk 3 above)
Comment (davidlougheed) [reply chord_metadata_service/patients/api_views.py:137]: yes it is

**Review by v-rocheleau — APPROVED** [review]
Works with bento_public PR, lgtm!

**Review by gsfk — APPROVED** [review]
Works fine. I guess there's no better filename than `<ugly_id>.json`?

**Pull request merged**